visualization/panda/panda3d_utils.py

```diff
@@ -0,0 +1,239 @@
+"""
+Panda3d Extension
+Author: Hao Chen
+"""
+import functools
+
+import numpy as np
+import cv2
+from panda3d.core import (Texture,
+                          NodePath,
+                          WindowProperties,
+                          Vec3,
+                          Point3,
+                          PerspectiveLens,
+                          OrthographicLens,
+                          PGTop,
+                          PGMouseWatcherBackground)
+from direct.gui.OnscreenImage import OnscreenImage
+
+import visualization.panda.filter as flt
+import visualization.panda.inputmanager as im
+import visualization.panda.world as wd
+
+
+def img_to_n_channel(img, channel=3):
+    """
+    Repeat a channel n times and stack
+    :param img:
+    :param channel:
+    :return:
+    """
+    return np.stack((img,) * channel, axis=-1)
+
+
+def letterbox(img, new_shape=(640, 640), color=(114, 114, 114), auto=True, scaleFill=False, scaleup=True, stride=32):
```

Comment:
1. Use letter_box, scale_fill, scale_up
2. Use rgb or rgba  instead of color. Each color value needs to be normalized to 0~1 instead of the raw 255
3. Write a comment with date and your names

Reply:
1 and 2 Updated. However, this function is copied from YOLOv5. I add a comment for it